### PR TITLE
Formatting, throwing new Exceptions, adding TaxOverride methods.

### PR DESCRIPTION
**Pull request by zfrolov**
Please do not merge until I remove the TODOs in the TaxOverride functions. I need to verify the constraints we should enforce after speaking to Manas and/or Tracy.

* Created line-level TaxOverride functionality.
* Added checks to functions that modify a line to ensure there is a line to work on.
* Moved some functions around to maintain order.
* Changed commit code to allow user to set their desired flag.
* Auto-formatted code.
* Minor addition to the example program to use TaxOverrides.

### CODE REVIEW

File: clients/dotnet/TransactionExtensions.cs [hunk 1]
```diff
@@ -38,12 +38,12 @@ public TransactionBuilder(AvaTaxClient client, string companyCode, DocumentType
 
         #region Builder Pattern
         /// <summary>
-        /// Flag this transaction to commit
+        /// Set the commit flag of the transaction.
         /// </summary>
         /// <returns></returns>
-        public TransactionBuilder WithCommit()
+        public TransactionBuilder WithCommitFlag(bool? commit)
```
Comment (ted-spence-avalara) [inline clients/dotnet/TransactionExtensions.cs]: Shouldn't this default to true?  In general the commit flag starts as false.  The only reason I can imagine calling this method is to flip it to true.

File: clients/dotnet/TransactionExtensions.cs [hunk 2]
```diff
@@ -76,6 +76,12 @@ public TransactionBuilder WithDiscountAmount(decimal? discount)
         /// <returns></returns>
         public TransactionBuilder IsItemDiscounted(bool? discounted)
```
Comment (ted-spence-avalara) [inline clients/dotnet/TransactionExtensions.cs]: I notice that the name of this function is odd.  It is "IsItemDiscounted" - shouldn't this be "WithItemDiscount"?

File: clients/dotnet/TransactionExtensions.cs [hunk 3]
```diff
@@ -184,9 +196,15 @@ public TransactionBuilder WithLatLong(TransactionAddressType type, decimal latit
         /// <returns></returns>
         public TransactionBuilder WithLineAddress(TransactionAddressType type, string line1, string line2, string line3, string city, string region, string postalCode, string country)
         {
+            // Ensure this can only be invoked when a line has been created.
+            if (_model.lines.Count <= 0)
+            {
+                throw new Exception("No lines have been added yet.");
+            }
+
```
Comment (ted-spence-avalara) [inline clients/dotnet/TransactionExtensions.cs]: I'm noticing that this is repeated many times.  Let's break this out into a separate function called GetMostRecentLine() which throws this error message consistently.

**Comment by zfrolov** [comment]
Curious about the efficacy of using reflection to get the caller method names. Are there compatibility issues?

**Comment by ted-spence-avalara** [comment]
@zfrolov - We'd have to test this library in both C# 2.0 and DotNetStandard to ensure it works in both cases.  If we have trouble, we can switch back to just using string identifiers.  I'm not totally sure it's necessary to identify the function name, as long as the user gets a clear error immediately when using it without any defined lines.

**Comment by ted-spence-avalara** [comment]
Reviewed with Zhenya and unblocked the merge

**Pull request merged**